src/Atmos/Model/AtmosModel.jl

```diff
@@ -537,21 +537,15 @@ equations.
     direction,
 )
 
-    vec_pad = SVector(1, 1, 1)
-    tens_pad = SArray{Tuple{3, 3}}(ntuple(i -> 1, 9))
     tend = Flux{FirstOrder}()
     _args = (; state, aux, t, direction)
     args = merge(_args, (precomputed = precompute(atmos, _args, tend),))
-    flux.ρ = Σfluxes(eq_tends(Mass(), atmos, tend), atmos, args) .* vec_pad
-    flux.ρu =
-        Σfluxes(eq_tends(Momentum(), atmos, tend), atmos, args) .* tens_pad
-
-    flux_first_order!(atmos.energy, atmos, flux, args)
-    flux_first_order!(atmos.moisture, atmos, flux, args)
-    flux_first_order!(atmos.precipitation, atmos, flux, args)
-    flux_first_order!(atmos.tracers, atmos, flux, args)
-    flux_first_order!(atmos.turbconv, atmos, flux, args)
 
+    map(prognostic_vars(atmos)) do prog
```

Comment:
`foreach` might be a better choice here?

```suggestion
    foreach(prognostic_vars(atmos)) do prog
```

Reply:
Interesting. Switching from `map` to `foreach` resulted in `getindex` at `var, name = get_prog_state`. Maybe this is a potential improvement to julialang? (assuming it's not already been improved in 1.6)